application_sdk/inputs/sql_query.py

```diff
@@ -63,9 +63,38 @@ def _read_sql_query(
         """
         import pandas as pd
         from sqlalchemy import text
+        from sqlalchemy.sql.elements import TextClause
 
         conn = session.connection()
-        return pd.read_sql_query(text(self.query), conn, chunksize=self.chunk_size)
+        try:
+            # First try using SQLAlchemy's execute
+            if isinstance(self.query, TextClause):
```

Comment:
Why is this condition needed? Not sure if this is the correct way to handle the error